.github/workflows/swift.yml

```
@@ -0,0 +1,91 @@
+# This workflow will build a Swift project
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-swift
+
+name: Swift
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+
+env:
+  SWIFT_VERSION: '5.9'
+  SWIFT_FORMAT_VERSION: '509.0.0'
+
+jobs:
+  build:
+
+    runs-on: macos-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - uses: swift-actions/setup-swift@v1
+      with:
+        swift-version: ${{ env.SWIFT_VERSION }}
+    - name: Restore dependencies cache
+      uses: actions/cache@v3
+      with:
+        path: |
+          .build/repositories
+          .build/checkouts
+        key: ${{ runner.os }}-swiftpm-${{ hashFiles('**/Package.*') }}
+
+    - name: Install dependencies
+      run: swift package resolve
+    - name: Build
+      run: swift build -v
+    #- name: Run tests
+    #  run: swift test -v
+
+  reviewdog-format:
+
+    name: Formatting
+    if: github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
```

```suggestion
    runs-on: ubuntu-latest
    permissions:
      contents: read
      pull-requests: write
```

add permission for `reviewdog/action-suggester`